pkg/exchange/binance/stream.go

```diff
@@ -146,34 +163,63 @@ func (s *Stream) handleDisconnect() {
 }
 
 func (s *Stream) handleConnect() {
+	var err error
 	if !s.PublicOnly {
-		// Emit Auth before establishing the connection to prevent the caller from missing the Update data after
-		// creating the order.
+		if s.ed25519Auth {
+			err = s.authConnectEd25519()
+		} else {
+			log.Warnf("Ed25519 private key is not set, using deprecated HMAC authentication")
+			err = s.authConnectHMAC()
+		}
+	} else {
+		err = s.subscribe()
+	}
+	if err != nil {
+		log.WithError(err).Error("subscribe error")
+	}
+}
+
+func (s *Stream) authConnectHMAC() error {
+	go s.EmitAuth()
+	return nil
+}
 
-		// spawn a goroutine to emit auth event to prevent blocking the main event loop
-		go s.EmitAuth()
-		return
+func (s *Stream) authConnectEd25519() error {
+	timestamp := time.Now().UnixNano() / 1e6
+	params := map[string]string{
+		"apiKey":    s.client.APIKey,
+		"timestamp": fmt.Sprintf("%d", timestamp),
```

Comment:
always prefer using strconv.FormatInt instead of Sprintf when you can use trconv.FormatInt

Reply:
Why? Is there a performance or security issue on it?